### PR TITLE
[CDAP-16663] Properly surface macros in input schema

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Properly generate input schema if previous stages' output schema is a macro
- Fix input schema directive to properly handle macro (including multiple input schemas)

JIRA: https://issues.cask.co/browse/CDAP-16663
Build: https://builds.cask.co/browse/CDAP-URUT240-1

### CODE REVIEW

File: cdap-ui/app/directives/widget-container/widget-input-schema/widget-input-schema.html [hunk 1]
```diff
@@ -52,6 +53,12 @@
           errors="errors[inputschema.name]">
         </my-complex-schema>
       </div>
+      <div
```
Comment (yukiej) [inline cdap-ui/app/directives/widget-container/widget-input-schema/widget-input-schema.html:56]: Should the comments at the beginning of this file be updated to reflect the changes?

File: cdap-ui/app/directives/widget-container/widget-input-schema/widget-input-schema.html [hunk 2]
```diff
@@ -52,6 +53,12 @@
...
+        ng-if="inputschema.isMacro"
+        ng-class="{'hide': MyInputSchemaCtrl.currentIndex !== $index, '': MyInputSchemaCtrl.currentIndex === $index}"
```
Comment (yukiej) [inline cdap-ui/app/directives/widget-container/widget-input-schema/widget-input-schema.html]: Possibly dumb question: Why do we need the second part of this expression in ng-class?

**Comment by yukiej** [comment]
Could we add an integration test for this?

File: cdap-ui/app/directives/widget-container/widget-input-schema/widget-input-schema.html (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/directives/widget-container/widget-input-schema/widget-input-schema.html]: Thats not a dumb question at all. Dumb code which needs to be cleaned up. 🤦

**Review by yukiej — APPROVED** [review]
lgtm!

**Comment by ajainarayanan** [comment]
Green build: https://builds.cask.co/browse/CDAP-URUT240-1

**Pull request merged**